tests/ui/methods.rs

```diff
@@ -385,6 +385,16 @@ fn iter_skip_next() {
     let _ = foo.filter().skip(42).next();
 }
 
+/// Checks implementation of the `FOLD_ANY` lint
+fn fold_any() {
+    let _ = (0..3).fold(false, |acc, x| acc || x > 2);
+}
+
+/// Checks implementation of the `FOLD_ANY` lint
+fn fold_any_ignore_initial_value_of_true() {
+    let _ = (0..3).fold(true, |acc, x| acc || x > 2);
+}
```

Comment:
You might also want to add another test that doesn't trigger the lint because it's not using a boolean, e.g. `.fold(0, |acc, x| acc += if x > 2 { 1 } else { 0 })`

Reply:
Will do.